pkg/node_service/node_service_client.go

```diff
@@ -12,31 +12,43 @@ import (
 	"k8s.io/klog/v2"
 )
 
-// Connect to the node_service gRPC server at the given address and retrieve initiators
-func GetNodeInitiators(nodeAddress string, reqType pb.InitiatorType) ([]string, error) {
+func InitializeClient(nodeAddress string) (conn *grpc.ClientConn, err error) {
 	port, envFound := os.LookupEnv(common.NodeServicePortEnvVar)
 	if !envFound {
 		port = "978"
 		klog.InfoS("no node service port found in environment. using default", "port", port)
 	}
-
 	nodeServiceAddr := nodeAddress + ":" + port
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-
-	conn, err := grpc.Dial(nodeServiceAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	conn, err = grpc.Dial(nodeServiceAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	if err != nil {
 		klog.ErrorS(err, "Error connecting to node service", "node ip", nodeAddress, "port", port)
-		return nil, err
+		return
 	}
-	defer conn.Close()
+	return
+}
 
+// Connect to the node_service gRPC server at the given address and retrieve initiators
+func GetNodeInitiators(conn *grpc.ClientConn, reqType pb.InitiatorType) ([]string, error) {
 	client := pb.NewNodeServiceClient(conn)
 	initiatorReq := pb.InitiatorRequest{Type: reqType}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Instead of creating a new context via `context.Background()`, you should consider passing in `ctx context.Context` so that you inherit the log level and any other key=value pairs stored in the context.